### PR TITLE
Add PySide2

### PR DESCRIPTION
**Pull request by jschueller**

### CODE REVIEW

**Comment by fredrikaverpil** [comment]
### PYSIDE-552

There's a bug ([PYSIDE-552](https://bugreports.qt.io/browse/PYSIDE-552)) preventing QtUiTools to be built and I can also see that module is not listed [here](https://github.com/conda-forge/staged-recipes/pull/3765/commits/b766da15d35d77142f2bb7e5fb766436c12eb95e#diff-1716e70ae6e69afa1c6c3d361ef0a671R37). This can be worked around using a `sed` command:

```bash
# Linux
sed -i.bak $'s/if(Qt5Designer_FOUND)/find_package(Qt5Designer)\\\nif(Qt5Designer_FOUND)/g' pyside-setup/sources/pyside2/CMakeLists.txt
```

### PYSIDE-357

There's another bug ([PYSIDE-357](https://bugreports.qt.io/browse/PYSIDE-357)) which causes `pyside2-uic` to not work. This too can be remedied by a `sed` command. Apparently, the `setup.py` script doesn't follow today's convention... and won't copy a folder tree. Instead its copy function only copies the files in the given directory and this causes subdirectories to not be copied. Here's a ugly, but working, workaround using sed (it's ugly because we're explicitly defining inclusions instead of exclusions):

```bash
# Linux
sed -i -e "s~\b\(packages\b.*\)],~\1, 'pyside2uic.Compiler', 'pyside2uic.port_v' + str(sys.version_info[0])],~" pyside-setup/setup.py
```

I've [asked](https://gitter.im/PySide/pyside2?at=599e59a4a7b406262dbf2e7a) the upstream (PySide2) developers to fix this, as we've pinpointed the issues and have working fixes. But development seems to be going slowly just in general. So I think it would be very valuable to have these things working in this conda-distributed build.

### Version

The Qt Company does not maintain the `PySide2.__version__` string. I requested them to implement some kind of version check which could e.g. be based on commit and/or date. They ended up implementing [this](http://code.qt.io/cgit/pyside/pyside-setup.git/commit/?h=5.6&id=8aca0dec116c6b321373f5636b34036530ac6e4b).

However, when checking my installation, I can't see all those attributes. This is what I get:

```python
__version__         = "2.0.0~alpha0"
__version_info__    = (2, 0, 0, "alpha", 0)

__build_date__ = '2017-08-21T11:46:13+00:00'

__build_commit_hash__ = '8aca0dec116c6b321373f5636b34036530ac6e4b'
__build_commit_hash_described__ = '2.0.0.dev0-4855-g8aca0de'
```

As you can see, we're dealing with `5.6`, `2.0.0~alpha0`, `2.0.0.dev0` as possible versions. I don't believe anyone has really decided what version PySide2 is actually on at the moment. These values have just been added over time with no consistency. I guess `5.6` refers to the version of Qt this version of PySide2 is intended to be built against.


### Git tag

I see you're using a [git tag](https://github.com/conda-forge/staged-recipes/pull/3765/commits/b766da15d35d77142f2bb7e5fb766436c12eb95e#diff-1716e70ae6e69afa1c6c3d361ef0a671R9) `5.6`. Would it not be better to check out the `5.6` branch?

**Comment by jschueller** [comment]
@fredrikaverpil 
- I included your patch for 552
- I also included https://bugreports.qt.io/browse/PYSIDE-562
- I went for version to 2.0.0.dev0 as it was recently changed:
"""
-__version__ = "5.6"
+__version__ = "2.0.0.dev0"
"""
- git tag 5.6 actually checks out the 5.6 branch

**Comment by fredrikaverpil** [comment]
Some bugs were fixed [upstream](http://code.qt.io/cgit/pyside/pyside-setup.git/log/?h=5.6):

- http://code.qt.io/cgit/pyside/pyside-setup.git/commit/?h=5.6&id=117e0ff91275b4bc06dd5383f19e7028c5ef6ff8
- http://code.qt.io/cgit/pyside/pyside-setup.git/commit/?h=5.6&id=c14949413fac467a94ba7dbc0e613fa1c3594358

**Comment by jschueller** [comment]
I know, see edits.

**Comment by jschueller** [comment]
This is ready to ship (dont mind the timeouts), if someone from @conda-forge/core agrees ?

**Comment by scopatz** [comment]
Don't the timeouts mean that it actually couldn't build and that now package would be uploaded even if we merged?

**Comment by jschueller** [comment]
@scopatz no, as for travis there would be several jobs, and for appveyor they're just connection errors.

**Comment by scopatz** [comment]
I'm not sure what to do in this situation...

**Comment by fredrikaverpil** [comment]
Is timeouts the problem here?
Can't this be solved like with [qt-feedstock](https://github.com/conda-forge/qt-feedstock)?
I'm sure their build times are enormous. Looks like their builds are completely disabled.

File: recipes/pyside2/bld.bat [hunk 1]
```diff
@@ -0,0 +1,56 @@
+
+set CMAKE_CONFIG="Release"
+
+cd %SRC_DIR%\sources\shiboken2
+mkdir build && cd build
+
+cmake -LAH -G"NMake Makefiles"                               ^
```
Comment (isuruf) [inline recipes/pyside2/bld.bat]: How about using `Ninja` to build parallelly?

Comment (jschueller) [reply recipes/pyside2/bld.bat]: It has no advantages over make -jX. Besides it has also limitations on how rules depend on each other/explicit rules, I already seen this for openturns.

Comment (isuruf) [reply recipes/pyside2/bld.bat]: I meant in the windows build, `nmake` builds serially right?

**Comment by jschueller** [comment]
@scopatz, see appveyor builds fine when there's no connection error and travis has time to build py27 and py35 so it will be fine in the feedstock.

File: recipes/pyside2/yum_requirements.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+libX11-devel
+libXt-devel
+libXext-devel
+libXrender-devel
+gtk2-devel
```
Comment (ocefpaf) [inline recipes/pyside2/yum_requirements.txt]: All but gtk are available as conda packages now. (And hopefully gtk will be available soon as well.)
Could you try using those? (No need to do it here, that experiment can happen on the feedstock.)

Comment (jschueller) [reply recipes/pyside2/yum_requirements.txt]: ok I'll look into this, thanks

**Pull request merged**